types/dom-background-sync/index.d.ts

```diff
@@ -0,0 +1,20 @@
+// Type definitions for non-npm package dom-background-sync-browser 0.0
+// Project: https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API
+// Definitions by: Hana Joo <https://github.com/h-joo>
+//                 Jan Kuehle <https://github.com/frigus02>
+//                 Martin Probst <https://github.com/mprobst>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.4
+
+interface SyncEventInit extends ExtendableEventInit {
+    lastChance?: boolean;
```

Comment:
I wonder if we should use `boolean | undefined` here. I think the TypeScript team has been doing this to better support https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes.

Reply:
Well actually according to :https://developer.mozilla.org/en-US/docs/Web/API/SyncEvent/SyncEvent, this should be non boolean only. WDYT?

Reply:
It's an optional boolean, yeah. As far as I understand TypeScript allows undefined for optional properties, except when exactOptionalPropertyTypes is enabled.

Most libraries in DefinitelyTyped were recently changed to use `| undefined`, e.g.: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54357

I just tested this. Chrome creates the same SyncEvent for `{ tag: '' }` and `{ tag: '', lastChance: undefined }`.

Reply:
As I wrote above, Chrome seems to treat `lastChance: undefined` in the same way as not specifying `lastChance` at all. This suggests `boolean | undefined` is okay here.